app/controllers/posts_controller.rb

```diff
@@ -23,6 +23,14 @@ def create
     end
   end
 
+  def destroy
+    @post = Post.find(params[:id])
+    @author = @post.author
+    @author.decrement!(:posts_counter!)
+    @post.destroy!
+    redirect_to user_posts_path(id: @author.id), notice: 'Post successfully deleted.'
+  end
+
   private
 
```

Comment:
- [ ] Good job so far, however, let's also ensure that CanCanCan is used for Post authorization in the _Post Controller_ so as to allow authorization of resources. We can achieve this by adding helper methods like the example below:

```ruby
  class PostsController < ApplicationController
    load_and_authorize_resource

    ...
  end
```